src/backend/distributed/deparser/deparse_table_stmts.c

```diff
@@ -273,6 +302,85 @@ AppendAlterTableCmdAddConstraint(StringInfo buf, Constraint *constraint,
 			appendStringInfo(buf, " NO INHERIT");
 		}
 	}
+	else if (constraint->contype == CONSTR_FOREIGN)
+	{
+		appendStringInfoString(buf, " FOREIGN KEY");
+
+		AppendColumnNameList(buf, constraint->fk_attrs);
+
+		appendStringInfoString(buf, " REFERENCES");
+
+		appendStringInfo(buf, " %s", quote_identifier(constraint->pktable->relname));
+
+		if (list_length(constraint->pk_attrs) > 0)
+		{
+			AppendColumnNameList(buf, constraint->pk_attrs);
+		}
+
+		/*
+		 * Append supported options if provided.
+		 * Note: When a constraint name is provided by the client therefore we do not deparse the statement in preprocessing,
+		 * the unsupported statements are caught in postprocessing in ErrorIfUnsupportedForeignConstraintExists function.
+		 * Hence the error messages for the unsupported alter table statements are unexpectedly different for the
+		 * ALTER TABLE ... ADD FOREIGN KEY and ALTER TABLE ...  ADD CONSTRAINT <conname> FOREIGN KEY
+		 */
+
+		/* FKCONSTR_MATCH_SIMPLE is default. Append matchtype if not default */
+		if (constraint->fk_matchtype == FKCONSTR_MATCH_FULL)
+		{
+			appendStringInfoString(buf, " MATCH FULL");
+		}
+
+		switch (constraint->fk_del_action)
+		{
+			case FKCONSTR_ACTION_NOACTION:
+			{
+				appendStringInfoString(buf, " ON DELETE NO ACTION");
+				break;
+			}
+
+			case FKCONSTR_ACTION_RESTRICT:
+			{
+				appendStringInfoString(buf, " ON DELETE RESTRICT");
+				break;
+			}
+
+			case FKCONSTR_ACTION_CASCADE:
+			{
+				appendStringInfoString(buf, " ON DELETE CASCADE");
+				break;
+			}
+
+			default:
```

Comment:
Why dont we have FKCONSTR_ACTION_SETDEFAULT, FKCONSTR_ACTION_SETNULL here?

Reply:
Similar to this example, https://github.com/citusdata/citus/pull/6616#discussion_r1081221501
you can do `ON DELETE SET NULL`, `ON DELETE SET DEFAULT`